src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -1087,6 +1087,8 @@ namespace cryptonote
   //---------------------------------------------------------------
   bool parse_and_validate_block_from_blob(const blobdata& b_blob, block& b)
   {
+    if(b_blob.size()==0)
```

Comment:
remove it

Reply:
It's essential otherwise
```
018-11-14 12:27:43.740   0x7fffb75e8380        ERROR   cn      src/cryptonote_basic/cryptonote_format_utils.cpp:1096   Failed to parse block from blob
```

Reply:
place it outside of function. you can't change invariant of base function